sources/editor/Stride.Assets.Presentation/AssetEditors/AssetCompositeGameEditor/ViewModels/AssetCompositeHierarchyEditorViewModel.cs

```diff
@@ -137,11 +137,10 @@ public override void Destroy()
         [CanBeNull]
         public override IEditorGamePartViewModel FindPartViewModel(AbsoluteId id)
         {
-            var item = RootPart as IEditorGamePartViewModel;
-            if (item != null && id == item.Id)
+            if (RootPart is IEditorGamePartViewModel item && id == item.Id)
                 return item;
-
-            return RootPart?.EnumerateChildren().BreadthFirst(x => x.EnumerateChildren()).OfType<IEditorGamePartViewModel>().FirstOrDefault(part => part.Id == id);
+            
+            return RootPart?.EnumerateChildren().BreadthFirst(x => x.EnumerateChildren()).FirstOrDefault(part => part is IEditorGamePartViewModel viewModel && viewModel.Id == id) as IEditorGamePartViewModel;
```

Comment:
I could be wrong, but it seems to me this version will do one more cast that the previous one.

Could be interesting to look at the generated IL to confirm.

Reply:
Again, OfType creates unnecessary allocations and that's the main issue we're trying to solve here.